apps/chat-e2e/src/testData/conversationHistory/conversationData.ts

```diff
@@ -215,17 +215,18 @@ export class ConversationData extends FolderData {
       conversation,
       replayIndex,
     );
+    const conversationCopy = JSON.parse(
+      JSON.stringify(conversation),
+    ) as Conversation;
     const partialAssistantMessage = replayIndex
-      ? conversation.messages[replayIndex + 2]
-      : conversation.messages.findLast((m) => m.role === 'assistant');
+      ? conversationCopy.messages[replayIndex + 2]
+      : conversationCopy.messages.findLast((m) => m.role === 'assistant');
```

Comment:
I know that this is not what you have changed, but why do we have +2 index here? Probably we need some comments?

Reply:
left comments